base/i18n/icu_util.cc

```diff
@@ -86,15 +86,19 @@ void LazyInitIcuDataFile() {
   if (g_icudtl_pf != kInvalidPlatformFile) {
     return;
   }
-#if defined(OS_ANDROID)
-  int fd = base::android::OpenApkAsset(kAndroidAssetsIcuDataFileName,
-                                       &g_icudtl_region);
-  g_icudtl_pf = fd;
-  if (fd != -1) {
-    return;
-  }
-// For unit tests, data file is located on disk, so try there as a fallback.
-#endif  // defined(OS_ANDROID)
+//It's not correct to check and read from assets folder first, since xwalk
+//put icudtl.dat in other folders like res/raw instead of assets.Reading
```

Comment:
And comments' in ResourceExtractor.java

"// Loading "icudtl.dat" from "assets/"" currently does not work with either
// embedded mode (the file is in raw/res) or shared mode (the app's context is
// used to retrieve the AssetManager, not Crosswalk's). We thus need to put
// those special files in a different directory so that we leverage the fallback
// code in Chromium to load these files from disk